components/eventing-controller/pkg/handlers/nats_test.go

```diff
@@ -957,20 +1012,28 @@ func TestIsValidSubscription(t *testing.T) {
 
 func TestSubscriptionUsingCESDK(t *testing.T) {
 	g := NewWithT(t)
+	defaultLogger := getLogger(g, kymalogger.INFO)
+
 	natsServer, _ := startNATSServer()
 	defer natsServer.Shutdown()
-	defaultLogger := getLogger(g, kymalogger.INFO)
+
 	natsConfig := env.NatsConfig{
-		URL:           natsServer.ClientURL(),
-		MaxReconnects: 2,
-		ReconnectWait: time.Second,
+		URL:                 natsServer.ClientURL(),
+		MaxReconnects:       2,
+		ReconnectWait:       time.Second,
+		MaxIdleConns:        50,
+		MaxConnsPerHost:     50,
+		MaxIdleConnsPerHost: 2,
+		IdleConnTimeout:     6 * time.Second,
 	}
 	defaultMaxInflight := 1
 	defaultSubsConfig := env.DefaultSubscriptionConfig{MaxInFlightMessages: defaultMaxInflight}
 	natsBackend := NewNats(natsConfig, env.DefaultSubscriptionConfig{MaxInFlightMessages: defaultMaxInflight}, metrics.NewCollector(), defaultLogger)
 	g.Expect(natsBackend.Initialize(nil)).Should(Succeed())
 
-	subscriber := eventingtesting.NewSubscriber()
+	subscriber := eventingtesting.NewSubscriber(
```

Comment:
Why not make the `eventingtesting.WithCloudEventServeMux()` part of the `NewSubscriber` constructor, since we want the same logic every time ?

Reply:
No, some of the old test still need the old serveMux. For that reason I added `WithCloudEventServeMux()` as an option only for the case that CloudEvents are explicitly needed.